handlers.go

```diff
@@ -1105,9 +1105,37 @@ func (s *Server) dealStatusByID(ctx context.Context, dealid uint) (*dealStatus,
 type getContentResponse struct {
 	Content      *Content       `json:"content"`
 	AggregatedIn *Content       `json:"aggregatedIn,omitempty"`
+	Selector     string         `json:"selector,omitempty"`
 	Deals        []*contentDeal `json:"deals"`
 }
 
+func (s *Server) calcSelector(aggregatedIn uint, contentID uint) (string, error) {
+	// sort the known content IDs aggregated in a CAR, and use the index in the sorted list
+	// to build the CAR sub-selector
+	rows, err := s.DB.Table("contents").
+		Select("id").
+		Where("aggregated_in = ?", aggregatedIn).
+		Order("id asc").
+		Rows()
+	defer rows.Close()
+
+	if err != nil {
+		return "", err
+	}
+
+	index := 0
+	for rows.Next() {
+		var id uint
+		rows.Scan(&id)
+		if id == contentID {
+			return fmt.Sprintf("/Links/%d/Hash", index), nil
+		}
+		index += 1
+	}
+
+	return "", nil
+}
+
```

Comment:
@marshall there is a gnarly nuance here: if you order by *numeric* ID you will get incorrect ordering on "number of digits in id" changes, since the directory assembler performs lexicographic sort over the filenames. So you want to `id::TEXT` the sort. 

But more to the point: the entire thing can be very efficiently be done in-rdbms, so I would just do something like:
```
SELECT ordinal - 1
FROM (
  SELECT
    id,
  ROW_NUMBER() OVER ( ORDER BY id::TEXT ) AS ordinal
  FROM contents
  WHERE aggregated_in = 9961183
) subq
WHERE id = 9961110
```

Reply:
ahh yes, you mentioned that the first time we talked about it and I had forgotten the nuance. thank you for reminding me and the SQL code example

Reply:
I think we should probably remove the `sort` call in the exising `createAggregate` method, we're sorting by ID there which gives the reader the impression that things will be ordered numerically, not text based.